src/me/remie/vulcan/leaguetasks/utils/LeaguePanel.java

```diff
@@ -0,0 +1,123 @@
+package me.remie.vulcan.leaguetasks.utils;
+
+import simple.hooks.simplebot.Game;
+import simple.hooks.wrappers.SimpleWidget;
+import simple.robot.api.ClientContext;
+
+public class LeaguePanel {
+    private ClientContext ctx;
+    private static final int WIDGET_ID = 657;
+
+    public LeaguePanel(final ClientContext ctx) {
+        this.ctx = ctx;
+    }
+
+    public boolean doWeNeedToClearFilters() {
+        final SimpleWidget tierDropDownMenu = ctx.widgets.getWidget(WIDGET_ID, LeaguePanelFilterType.FILTER_TIER.getWidgetId());
+        if (!isOptionSelected(tierDropDownMenu)) {
+            return true;
+        }
+
+        final SimpleWidget typeDropDownMenu = ctx.widgets.getWidget(WIDGET_ID, LeaguePanelFilterType.FILTER_TYPE.getWidgetId());
+        if (!isOptionSelected(typeDropDownMenu)) {
+            return true;
+        }
+
+        final SimpleWidget areaDropDownMenu = ctx.widgets.getWidget(WIDGET_ID,LeaguePanelFilterType.FILTER_AREA.getWidgetId());
+        if (!isOptionSelected(areaDropDownMenu)) {
+            return true;
+        }
+
+        final SimpleWidget completedDropDownMenu = ctx.widgets.getWidget(WIDGET_ID, LeaguePanelFilterType.FILTER_COMPLETED.getWidgetId());
+        if (!isOptionSelected(completedDropDownMenu)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    private boolean isOptionSelected(SimpleWidget widget) {
+        if (widget == null || widget.getChild(4) == null || widget.getChild(4).getText() == null) {
+            return false;
+        }
+        return widget.getChild(4).getText().equalsIgnoreCase("all");
+    }
+
+    private boolean chooseAllInDropDown(int filterWidgetId, int filterDropdownOptionId) {
+        final SimpleWidget filterDropDownMenu = ctx.widgets.getWidget(WIDGET_ID,filterWidgetId);
+        if (filterDropDownMenu != null) {
+            if (!isOptionSelected(filterDropDownMenu)) {
+                final SimpleWidget selectedOption = filterDropDownMenu.getChild(4);
+                final SimpleWidget dropDownMenu = ctx.widgets.getWidget(WIDGET_ID, filterDropdownOptionId);
+                if (!chooseOptionInDropdown(selectedOption, dropDownMenu)) {
+                    ctx.log("Something went wrong choosing all in options");
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
```

Comment:
```suggestion
    private boolean chooseAllInDropDown(int filterWidgetId, int filterDropdownOptionId) {
        final SimpleWidget filterDropDownMenu = ctx.widgets.getWidget(WIDGET_ID, filterWidgetId);
        if (filterDropDownMenu == null || isOptionSelected(filterDropDownMenu)) {
            return true;
        }
        final SimpleWidget selectedOption = filterDropDownMenu.getChild(4);
        final SimpleWidget dropDownMenu = ctx.widgets.getWidget(WIDGET_ID, filterDropdownOptionId);
        if (!chooseOptionInDropdown(selectedOption, dropDownMenu)) {
            ctx.log("Something went wrong choosing all in options");
            return false;
        }
        return true;
    }
```
I think early returns can improve code readability a lot. Just a nitpick honestly, can totally ignore this.